cmd/sign-off-checker/main.go

```diff
@@ -49,97 +45,62 @@ func main() {
 	if secretString == "" {
 		log.Fatal("SHARED_SECRET is not set")
 	}
-	secret = []byte(secretString)
 
 	token, _ := os.LookupEnv("GITHUB_TOKEN")
 	if token == "" {
 		log.Fatal("GITHUB_TOKEN is not set")
 	}
 
-	ts := oauth2.StaticTokenSource(
-		&oauth2.Token{AccessToken: token},
-	)
-	tc := oauth2.NewClient(oauth2.NoContext, ts)
-	client = github.NewClient(tc)
-
-	log.Print("Starting serving /webhook on :8080")
-	http.Handle("/webhook", loggingMiddleware(http.HandlerFunc(HandleHook)))
-	err := http.ListenAndServe(":8080", nil)
-	log.Fatal(err)
-}
-
-func HandleHook(w http.ResponseWriter, r *http.Request) {
-	payload, err := github.ValidatePayload(r, secret)
-	if err != nil {
-		http.Error(w,
-			fmt.Sprintf("Could not validate signature: %v", err),
-			http.StatusBadRequest)
-		return
+	autoRegisterOrgsString, _ := os.LookupEnv("AUTOREGISTER_ORGANIZATIONS")
+	autoRegisterOrgs := strings.Split(autoRegisterOrgsString, ",")
+	if autoRegisterOrgsString == "" {
+		autoRegisterOrgs = []string{}
 	}
 
-	hooktype := github.WebHookType(r)
-	event, err := github.ParseWebHook(hooktype, payload)
-	if err != nil {
-		http.Error(w,
-			fmt.Sprintf("Error parsing payload: %v", err),
-			http.StatusBadRequest)
-		return
-	}
-	switch event := event.(type) {
-	case *github.PullRequestEvent:
-		HandlePullRequest(event)
-	default:
-		log.Printf("Unhandled hook type: %v", hooktype)
+	publicWebhookURL, _ := os.LookupEnv("PUBLIC_WEBHOOK_URL")
+	if publicWebhookURL == "" && len(autoRegisterOrgs) > 0 {
+		log.Fatal("PUBLIC_WEBHOOK_URL is required for AUTOREGISTER_ORGANIZATIONS")
 	}
-}
 
-func HandlePullRequest(event *github.PullRequestEvent) {
-	owner := event.Repo.Owner.Login
-	repo := event.Repo.Name
-	number := event.Number
-
-	opt := &github.ListOptions{PerPage: 10}
-	allCommits := []*github.RepositoryCommit{}
-	for {
-		commits, resp, err := client.PullRequests.ListCommits(context.TODO(), *owner, *repo, *number, opt)
-		if err != nil {
-			log.Printf("Error getting commits for PR: %v", err)
-			return
-		}
-		allCommits = append(allCommits, commits...)
-		if resp.NextPage == 0 {
-			break
-		}
-		opt.Page = resp.NextPage
-	}
-
-	signMissing := false
-	for _, commit := range allCommits {
-		if !testRE.MatchString(*commit.Commit.Message) {
-			signMissing = true
-			break
-		}
-	}
+	dryRunString, _ := os.LookupEnv("DRY_RUN")
```

Comment:
Not documented in README.  Also makes me think that we should start moving to command line flags for this stuff.  The only reason other params were env variables is that it is bad form to put secrets on the command line.

Reply:
Sounds good. I'll pull in spf13/cobra + spf13/viper and update the command to use that.